### ISSUE TITLE
[SUGGESTION] Configurable Camera position for Spaceship 3rd person view

### ISSUE THREAD

**Issue by BonnyORG** [tsunamayo/Starship-EVO#756]
When Looking around in F4 in a ship, the center of orientation is always the ship center, instead of the player, as it should be.

Version: 18w15b

**Comment by Napther** [comment]
I would call this intentional, given that F4 acts like a freemode/orbit camera for the ship and would be pretty useless on huge vessels to orbit the player and be stuck inside the ship's confines.

**Comment by BonnyORG** [comment]
Still it does not look very intentional with the camera positioned so low when flying a freighter that's relatively high. The camera at the height of the bridge/player would look much better there

**Comment by tsunamayo** [comment]
Yes it was supposed to be intentional but placement is not perfect... Not sure on the best approach to solve this problem... If you have idea ;-)

**Comment by schlid** [comment]
The way I'd work around this would be by having a key to lock and unlock a free camera? So you can pick a camera position moving around with wasd and the mouse and then lock it in.

The key to enter free camera mode could be by holding f4 with a confirmation timer much like the one when you get into a chair or enter a system, so that there aren't additional keys being used unnecessarily.

From there, you would move the camera to the desired position, then left click with the mouse to confirm the camera position and resume controlling the ship, or right click to reset it to the standard position.

After a camera position was set, you'd effectively be in third person mode, so pressing F4 would toggle to and from the custom position of the camera to first person.

To reset the free camera to the normal third person position, holding f4 then pressing right click to reset to the normal third person camera position, then clicking to confirm this would bring everything back to normal.

Due to the hold down to access implementation, this wouldn't interfere with existing third person, and  would add on to it fluently.

**Comment by BonnyORG** [comment]
I'd simply fit the camera to the control block you have entered. This would also bring the possibility to add several control blocks that are facing sideways (eg. on big freighters) to be able to control the ship sideways for docking etc. without adding another ship Core that's facing sideways

**Comment by ZachZent** [comment]
Here is how I would do it. Press f4 once and you are in 3rd person where you see your chair. Like normal 3rd person while standing. Press it again and you are in a ship view. Only exception would be if you are using a system. Like a joystick

**Comment by JRL101** [comment]
I'd how it is, but add a function to **HOLD F4** while in 3rd person allows you to pull the camera back or forward with the scroll wheel. Although you could make a function in the "controller block/Ship block/mech controller/hover controller" to adjust the camera position. (either from the linked camera, or the default F4)

**Comment by tsunamayo** [comment]
Okay I still need to address this. I think I will add a button to the spaceship computer where we can configure the view offset. Thanks
